src/nrniv/nrncore_write/io/nrncore_io.cpp

```diff
@@ -390,25 +397,84 @@ void write_nrnthread_task(const char* path, CellGroup* cgs)
     /// Writing the file with task, correspondent number of threads and list of correspondent first gids
     if (nrnmpi_myid == 0)
     {
+        // If append is false, begin a new files.dat (overwrite old if exists).
+        // If append is true, append groupids to existing files.dat.
+        //   Note: The number of groupids (2nd or 3rd line) has to be
+        //   overwritten wih the total number so far. To avoid copying
+        //   old to new, we allocate 10 chars for that number.
+
         std::stringstream ss;
         ss << path << "/files.dat";
 
         std::string filename = ss.str();
 
-        FILE *fp = fopen(filename.c_str(), "w");
-        if (!fp) {
-            hoc_execerror("nrnbbcore_write write_nrnthread_task could not open for writing:", filename.c_str());
+        FILE *fp = NULL;
+        if (append == false) { // start a new file
+            fp = fopen(filename.c_str(), "w");
+            if (!fp) {
+                hoc_execerror("nrncore_write write_nrnthread_task could not open for writing:", filename.c_str());
+            }
+        } else { // modify groupid number and append to existing file
+            fp = fopen(filename.c_str(), "r+");
+            if (!fp) {
+                hoc_execerror("nrncore_write write_nrnthread_task could not open for modifying:",  filename.c_str());
+            }
+        } 
+
+        int maxlen = 20;
+        char line[maxlen]; // All lines are actually no larger than %10d.
+
+        if (append) {
+            // verify same version
+            fgets_err(line, maxlen, fp);
+            // unfortunately line has the newline
+            size_t n = strlen(bbcore_write_version);
+            if ((strlen(line) - 1 != n)
+                || strncmp(line, bbcore_write_version, n) != 0) {
+                fclose(fp);
+                hoc_execerror("nrncore_write bad existing files.dat version:", line);
+            }
+        } else {
+            fprintf(fp, "%s\n", bbcore_write_version);
         }
 
-        fprintf(fp, "%s\n", bbcore_write_version);
-
         // notify coreneuron that this model involves gap junctions
         if (nrnthread_v_transfer_) {
-            fprintf(fp, "-1\n");
+            if (append) {
+                fgets_err(line, maxlen, fp);
+                if (strcmp(line, "-1\n") != 0) {
+                    fclose(fp);
+                    hoc_execerror("nrncore_write existing files.dat does not have a gap junction indicator\n", NULL);
```

Comment:
same : message could be updated to tell user what could be done?

Reply:
I agree